### PR TITLE
Add sidecars tags to CF containers

### PR DESCRIPTION
**Pull request by NouemanKHAL**
### What does this PR do?

Adds `sidecar_present` and `sidecar_count` tags on Cloud Foundry containers that run apps with sidecar processes.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by therve — COMMENTED** [review]
Just a thought: what about including the sidecar info directly in the app payload, instead of adding an extra API endpoint? This way we get "for free" if the setting is present when getting the apps from the nozzle.

File: pkg/util/cloudproviders/cloudfoundry/types.go [hunk 1]
```diff
@@ -276,6 +301,10 @@ func DesiredLRPFromBBSModel(bbsLRP *models.DesiredLRP, includeList, excludeList
 			} else {
 				log.Debugf("Could not find org %s in cc cache", orgGUID)
 			}
+			if sidecars, err := ccCache.GetSidecars(appGUID); err == nil && len(sidecars) > 0 {
```
Comment (therve) [inline pkg/util/cloudproviders/cloudfoundry/types.go:304]: Should we put this behind a feature flag?

Comment (NouemanKHAL) [reply pkg/util/cloudproviders/cloudfoundry/types.go:304]: Yes we should, it makes sense to keep it optional (reduce api calls) and expose it under "Enable Cloud Foundry API apps polling" in the tile.

File: releasenotes/notes/pcf-sidecars-tags-90a54b31bbdacc67.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Adds `sidecar_present` and `sidecar_count` tags on Cloud Foundry containers
```
Comment (kayayarai) [inline releasenotes/notes/pcf-sidecars-tags-90a54b31bbdacc67.yaml:11]:
```suggestion
    Adds ``sidecar_present`` and ``sidecar_count`` tags on Cloud Foundry containers
```
Not 100% sure, but Agent release notes usually use double-backticks for monospace

**Review by kayayarai — APPROVED** [review]

File: releasenotes/notes/pcf-sidecars-tags-90a54b31bbdacc67.yaml (hunk 2 above)
Comment (NouemanKHAL) [reply releasenotes/notes/pcf-sidecars-tags-90a54b31bbdacc67.yaml:11]: You're right, thanks!

File: pkg/util/cloudproviders/cloudfoundry/types.go (hunk 1 above)
Comment (NouemanKHAL) [reply pkg/util/cloudproviders/cloudfoundry/types.go:304]: In this case, I'll tag start adding `sidecar_present:false` as well.

**Comment by NouemanKHAL** [comment]
Rebased the branch to fix some unrelated unit tests (will just merge main next time to avoid breaking history)

**Review by vboulineau — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by therve — APPROVED** [review]

**Pull request merged**